src/plopp/backends/matplotlib/figure.py

```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from .utils import is_interactive_backend
+
+
+class Fig1d:
```

Comment:
I guess it would work, but then I have to make the ones in plotly and pythreejs also functions...

Reply:
Not necessarily. The classes that use these only require a callable. This could be a function or a class. But it would be cleaner to use functions everywhere, IMHO.

Reply:
Something like

graphics/figure.py:
```Py
def figure1d(*args, style='line', **kwargs):

    if style == 'line':
        from .figline import FigLine

        return backends.figure1d(*args, FigConstructor=FigLine, **kwargs)
```

backends/manager.py:
```Py
class BackendManager:
...
    def figure1d(self, *args, **kwargs):
        try:
            return self._backends['2d'].figure1d(*args, **kwargs)
        except AttributeError:
            raise ValueError(f'Unsupported backend \'{self["2d"]}\' for figure1d.')
```

backends/matplotlib/\_\_init\_\_.py:
```Py

class MatplotlibBackend:

    def figure1d(self, *args, **kwargs):
        from .figure import figure1d as fig1d_mpl
        return fig1d_mpl(*args, **kwargs)
```

backends/matplotlib/figure.py:
```Py
def figure1d(*args, **kwargs):

    if is_interactive_backend():
        from .interactive import InteractiveFig1d
        return InteractiveFig1d(*args, **kwargs)
    else:
        from .static import StaticFig
        return StaticFig(*args, **kwargs)
```

?